src/openzaak/components/catalogi/api/validators.py

```diff
@@ -540,3 +546,66 @@ def __call__(self, attrs, serializer):
                 self.message.format(self.start_date_field, self.end_date_field),
                 code=self.code,
             )
+
+
+class ZaakTypeRelationsPublishValidator:
+    """
+        Validate that the ZaakType object has the correct relations for publishing
+        """
+
+    code = "overlap"
+    message = _(
+        "Dit {} komt al voor binnen de catalogus en opgegeven geldigheidsperiode."
+    )
+
+    requires_context = True
+
+    def __call__(self, attrs, serializer):
+        instance = getattr(serializer, "instance", None)
+
+        related_errors = dict()
+
+        # check related objects
+        if (
+            instance.besluittypen.filter(concept=True).exists()
+            or instance.informatieobjecttypen.filter(concept=True).exists()
+            or instance.deelzaaktypen.filter(concept=True).exists()
+        ):
+            msg = _("All related resources should be published")
+            related_errors[api_settings.NON_FIELD_ERRORS_KEY] = [msg]
+
+        has_invalid_resultaattypen = instance.resultaattypen.filter(
```

Comment:
Could you reuse `validate_zaaktype_for_publish` here?

Reply:
It should have been there already

Reply:
Done plus moved check for related unpublished objects their.